autogpt_platform/backend/backend/data/diagnostics.py

```diff
@@ -0,0 +1,1413 @@
+                "startedAt": None,
+            }
+        )
+
+        # Throughput metrics
+        completed_1h = await AgentGraphExecution.prisma().count(
+            where={
+                "executionStatus": AgentExecutionStatus.COMPLETED,
+                "updatedAt": {"gte": one_hour_ago},
+            }
+        )
+
+        completed_24h = await AgentGraphExecution.prisma().count(
+            where={
+                "executionStatus": AgentExecutionStatus.COMPLETED,
+                "updatedAt": {"gte": twenty_four_hours_ago},
+            }
+        )
+
+        throughput_per_hour = completed_24h / 24.0 if completed_24h > 0 else 0.0
+
+        return ExecutionDiagnosticsSummary(
+            running_count=running_count,
+            queued_db_count=queued_db_count,
+            rabbitmq_queue_depth=rabbitmq_queue_depth,
+            cancel_queue_depth=cancel_queue_depth,
+            orphaned_running=orphaned_running,
+            orphaned_queued=orphaned_queued,
+            failed_count_1h=failed_count_1h,
+            failed_count_24h=failed_count_24h,
+            failure_rate_24h=failure_rate_24h,
+            stuck_running_24h=stuck_running_24h,
+            stuck_running_1h=stuck_running_1h,
+            oldest_running_hours=oldest_running_hours,
+            stuck_queued_1h=stuck_queued_1h,
+            queued_never_started=queued_never_started,
+            invalid_queued_with_start=invalid_queued_with_start,
+            invalid_running_without_start=invalid_running_without_start,
+            completed_1h=completed_1h,
+            completed_24h=completed_24h,
+            throughput_per_hour=throughput_per_hour,
+            timestamp=now.isoformat(),
+        )
+    except Exception as e:
+        logger.error(f"Error getting execution diagnostics: {e}")
+        raise
+
+
+async def get_agent_diagnostics() -> AgentDiagnosticsSummary:
+    """
+    Get comprehensive agent diagnostics.
+
+    Returns:
+        AgentDiagnosticsSummary with agent metrics
+    """
+    try:
+        # Get distinct agent graph IDs with active executions
+        executions = await AgentGraphExecution.prisma().find_many(
+            where={
+                "executionStatus": {
+                    "in": [AgentExecutionStatus.RUNNING, AgentExecutionStatus.QUEUED]  # type: ignore
+                }
+            },
+            distinct=["agentGraphId"],
+        )
+
+        return AgentDiagnosticsSummary(
+            agents_with_active_executions=len(executions),
+            timestamp=datetime.now(timezone.utc).isoformat(),
+        )
+    except Exception as e:
+        logger.error(f"Error getting agent diagnostics: {e}")
+        raise
+
+
+async def get_schedule_health_metrics() -> ScheduleHealthMetrics:
+    """
+    Get comprehensive schedule diagnostics via Scheduler service.
+
+    Returns:
+        ScheduleHealthMetrics with schedule health info
+    """
+    try:
+        from backend.util.clients import get_scheduler_client
```

Comment:
the internal imports on each of the functions here can be top-level

Reply:
I did miss this one :(